src/HindBrain/HindBrain.ino

```diff
@@ -0,0 +1,374 @@
+  eStop = new Estop(&nh, ESTOP_SENSE_PIN, ESTOP_DEBOUNCE_TIME);
+  eStop->onStop(eStopTractor);
+  eStop->offStop(eStartTractor);
+  pinMode(ESTOP_RELAY_PIN, OUTPUT);
+  autoLight = new OAKSoftSwitch(&nh, "/auto_light", AUTO_LIGHT_PIN);
+
+  // Stop engine for safety
+  stopEngine();
+
+  // Open serial communication with roboclaw
+  rc1.begin(RC_BAUD_RATE);
+  rc2.begin(RC_BAUD_RATE);
+
+  // Set up ROS node, subscribers, publishers
+  nh.getHardware()->setBaud(SERIAL_BAUD_RATE);
+  nh.initNode();
+  nh.subscribe(driveSub);
+  nh.subscribe(hitchSub);
+  nh.subscribe(ping);
+  nh.subscribe(userInput);
+  nh.advertise(pubDrive);
+  nh.advertise(hitchPose);
+
+  // Wait for connection
+  while(true) {
+    if(nh.connected() && (millis() - watchdogTimer < WATCHDOG_TIMEOUT)) {
+      break;
+    }
+    nh.spinOnce();
+    delay(1);
+  }
+
+  nh.loginfo("Hindbrain connected, running startup calibration sequence");
+  delay(500);
+  runStartupSequence();
+  watchdogTimer = millis();
+}
+
+
+void loop() {
+  // Check for connectivity with mid-brain
+  checkSerial(&nh);
+
+  // Update current published pose
+  updateCurrDrive();
+  updateCurrHitchPose();
+
+  hitchMsg = computeHitchMsg();
+
+  // Send updated motor commands to roboclaws
+  if (!isEStopped) {
+    updateRoboClaw(velMsg, steerMsg, hitchMsg);
+  } else {
+    stopRoboClaw(&rc1, &rc2);
+  }
+
+  // Update node
+  nh.spinOnce();
+  delay(1);
+}
+
+
+void ackermannCB(const ackermann_msgs::AckermannDrive &msg) {
+  // Save steer and vel cmds to global vars.
+  steerMsg = steerAckToCmd(msg.steering_angle);
+  velMsg = velAckToCmd(msg.speed);
+}
+
+
+void hitchCB(const geometry_msgs::Pose &msg){
+  desiredHitchPose.position.z = msg.position.z; // In meters from flat ground
+  // TODO: Copy over all desired attributes
+}
+
+
+void watchdogCB(const std_msgs::Empty &msg) {
+  // Update watchdog timer on receipt of msg
+  watchdogTimer = millis();
+}
+
+
+void userInputCB(const std_msgs::String &msg) {
+  // Update input global var with msg only if msg is 1 char.
+  if (strlen(msg.data) == 1) {
+    usrMsg = *msg.data;
+    #ifdef DEBUG
+    char j[20];
+    snprintf(j, sizeof(j), "Received %s", msg.data);
+    nh.loginfo(j);
+    #endif
+  }
+}
+
+
+void waitForUserVerification() {
+  // Block main code until 'y' is received on /user_input.
+  while (true) {
+    auto cmd = checkUserInput();
+    if (cmd == 'y') {
+      break;
+    }
+    nh.spinOnce();
+    delay(1);
+  }
+}
+
+
+void runStartupSequence() {
+  // Run startup & calibration sequence with appropriate user input.
+  //TODO create prompt publishing topic
+
+  nh.loginfo("Remove pins on steering and velocity, then publish 'y' to /user_input topic");
+  waitForUserVerification();
+
+  nh.loginfo("Verification received, homing steering and velocity actuators . . .");
+  delay(500);
+  rc1.SpeedAccelDeccelPositionM1(RC1_ADDRESS, 0, 300, 0, velMsg, 1);
+  rc1.SpeedAccelDeccelPositionM2(RC1_ADDRESS, 0, 500, 0, steerMsg, 1);
```

Comment:
I'm scared that these velMsg, steerMsg, and hitchMsg are declared but uninitialized at this point.

Reply:
Yup, a race condition. Fixed.